core/src/main/scala/org/bitcoins/core/wallet/builder/TxBuilder.scala

```diff
@@ -402,68 +411,43 @@ sealed abstract class BitcoinTxBuilder extends TxBuilder {
         remaining: Seq[BitcoinUTXOSpendingInfo],
         currentLockTime: UInt32): Try[UInt32] = remaining match {
       case Nil => Success(currentLockTime)
-      case BitcoinUTXOSpendingInfo(outpoint,
-                                   output,
-                                   signers,
-                                   redeemScriptOpt,
-                                   scriptWitOpt,
-                                   hashType) :: t =>
-        output.scriptPubKey match {
-          case cltv: CLTVScriptPubKey =>
-            val lockTime =
-              if (cltv.locktime.toLong > UInt32.max.toLong || cltv.locktime.toLong < 0) {
-                TxBuilderError.IncompatibleLockTimes
-              } else Success(UInt32(cltv.locktime.toLong))
-            val result = lockTime.flatMap { l: UInt32 =>
-              if (currentLockTime < l) {
-                val lockTimeThreshold = tc.locktimeThreshold
-                if (currentLockTime < lockTimeThreshold && l >= lockTimeThreshold) {
-                  //means that we spend two different locktime types, one of the outputs spends a
-                  //OP_CLTV script by block height, the other spends one by time stamp
-                  TxBuilderError.IncompatibleLockTimes
-                } else Success(l)
-              } else Success(currentLockTime)
-            }
-            result
-          case _: P2SHScriptPubKey | _: P2WSHWitnessSPKV0 =>
-            if (redeemScriptOpt.isDefined) {
-              //recursively call with redeem script as output script
-              val o = TransactionOutput(output.value, redeemScriptOpt.get)
-              val i = BitcoinUTXOSpendingInfo(outpoint,
-                                              o,
-                                              signers,
-                                              None,
-                                              scriptWitOpt,
-                                              hashType)
-              loop(i +: t, currentLockTime)
-            } else if (scriptWitOpt.isDefined) {
-              scriptWitOpt.get match {
-                case EmptyScriptWitness    => loop(t, currentLockTime)
-                case _: P2WPKHWitnessV0    => loop(t, currentLockTime)
-                case p2wsh: P2WSHWitnessV0 =>
-                  //recursively call with the witness redeem script as the script
-                  val o = TransactionOutput(output.value, p2wsh.redeemScript)
-                  val i = BitcoinUTXOSpendingInfo(outpoint,
-                                                  o,
-                                                  signers,
-                                                  redeemScriptOpt,
-                                                  None,
-                                                  hashType)
-                  loop(i +: t, currentLockTime)
-              }
-            } else {
-              loop(t, currentLockTime)
+      case spendingInfo :: newRemaining =>
```

Comment:
just a note: we should probably use `+:` for pattern matching on collections as I believe `::` can throw a exception for certain types of collections. 

I know `::` is used elsewhere in this file -- it is unsafe and depends on the scala runtime "deciding" on what collections to use. We should be moving to use `+:`